PR (GEOS_7785) - Implement a REST Based Role Service for GeoServer

afabiani opened the pull request:
Releated to JIRA [GEOS-7785](https://osgeo-org.atlassian.net/browse/GEOS-7785)

afabiani commented:
Double checked the new dependencies included:

(*) The only new ones are:
- json-smart (118Kb)
- hancrest (44Kb)

(*) All the others are alredy dependencies of other plugins; in particular
- jettison (44Kb) already included in LDAP and GWC
- httpclient (716Kb) already included in gt-nectdf
- httpcore (320Kb) already included in gt-netcdf
- jackson-databind already included in many projects like hz-clister, gt-netcdf, geofence

simboss commented:
I see you are adding version in the plugin for some dependencies.

Can you check if these are aligned with the others and/or if there is something similar in the main pom?

afabiani commented:
Already checked, the versions are aligned.

I just moved the only three new ones signaled above into the main pom under "dependencyManagement".

With the last commit I also added user documentation.
